evm/src/witness/traces.rs

```diff
@@ -31,7 +31,7 @@ pub struct TraceCheckpoint {
 
 #[derive(Debug)]
 pub(crate) struct Traces<T: Copy> {
-    pub(crate) arithmetic_ops: Vec<arithmetic::Operation>,
+    pub(crate) arithmetic_ops: Vec<(arithmetic::Operation, bool)>,
```

Comment:
I'm not a fan of this tuple. Maybe it would be cleaner if it was at least put behind an alias or something?

Reply:
An alias would probably be an improvement. I haven't looked at this PR at all yet, so apologies if this is way off, but maybe the  `bool` should be integrated into the `arithmetic::Operation` type?

Reply:
I'll think about both approaches and come up with something in the afternoon!